jetty-core/jetty-xml/src/test/java/org/eclipse/jetty/xml/XmlConfigurationTest.java

```diff
@@ -1963,6 +1967,47 @@ public void testVirtualCallWithClassAttribute() throws Exception
         assertThat(tc.testObject, instanceOf(ScheduledExecutorService.class));
     }
 
+    public static Stream<Arguments> xmlSystemIdSource()
+    {
+        List<Arguments> ids = new ArrayList<>();
+
+        String[] schemes = {"http", "https"};
+        String[] hosts = {"eclipse.org", "www.eclipse.org", "eclipse.dev", "www.eclipse.dev"};
+        String[] paths = {"/jetty/configure.dtd", // version-less
+                          "/jetty/configure_9_0.dtd", // 9.0
+                          "/jetty/configure_9_3.dtd", // 9.3
+                          "/jetty/configure_10_0.dtd"}; // 10.
+
+        for (String scheme: schemes)
+        {
+            for (String host: hosts)
+            {
+                for (String path: paths)
+                {
+                    ids.add(Arguments.of(String.format("%s://%s%s", scheme, host, path)));
+                }
+            }
+        }
+
+        return ids.stream();
+    }
+
+    /**
+     * Test to ensure that all required XML System ID variants are covered in the
+     * {@link XmlConfiguration} internals.
+     */
+    @ParameterizedTest
+    @MethodSource("xmlSystemIdSource")
+    public void testSystemIdVariants(String xmlSystemId) throws IOException, SAXException
+    {
+        // empty raw xml, just to instantiate XmlConfiguration, so we can access the XmlParser / ConfigurationParser.
+        XmlConfiguration xmlConfiguration = asXmlConfiguration("<Configure class=\"org.eclipse.jetty.xml.TestConfiguration\" />");
+        XmlParser configurationProcessor = xmlConfiguration.getXmlParser();
+        XmlParser.Handler xmlHandler = configurationProcessor.newHandler();
+        InputSource inputSource = xmlHandler.resolveEntity(null, xmlSystemId);
```

Comment:
Can we avoid making `Handler` public, and its related getter, maybe adding `XmlParser.resolveEntity()` as package private only used by tests?